### PR TITLE
[Backtesting] Add --disable-backtesting-timeout arg to disable backtesting timeout

### PR DESCRIPTION
**Pull request by Herklos**

### CODE REVIEW

File: octobot/octobot_backtesting_factory.py [hunk 1]
```diff
@@ -34,14 +38,15 @@ async def initialize(self):
...
+                disable_join_timeout=self.disable_join_timeout)
             await octobot_backtesting_api.initialize_and_run_independent_backtesting(self.independent_backtesting,
                                                                                      log_errors=False)
             await octobot_backtesting_api.join_independent_backtesting(self.independent_backtesting)
             if self.log_report:
                 octobot_backtesting_api.log_independent_backtesting_report(self.independent_backtesting)
             await octobot_backtesting_api.stop_independent_backtesting(self.independent_backtesting, memory_check=False)
         except Exception as e:
-            self.logger.error(f"Error when starting backtesting: {e}")
+            self.logger.error(f"Error when starting backtesting: {e.__class__.__name__}")
```
Comment (GuillaumeDSM) [inline octobot/octobot_backtesting_factory.py:50]: 👍

File: octobot/cli.py [hunk 2]
```diff
@@ -232,6 +234,10 @@ def octobot_parser(parser):
                         help='On multiple files backtesting: run on the whole available data instead of the '
                              'common part only (default behavior).',
                         action='store_true')
+    parser.add_argument('-dbt', '--disable-backtesting-timeout',
+                        help='To use when running long backtesting : disable backtesting watcher timeout '
+                             'that prevent backtesting from being stuck during execution.',
```
Comment (GuillaumeDSM) [inline octobot/cli.py:239]:
```suggestion
                             'that prevent backtesting from being interrupted during execution.',
```

File: octobot/backtesting/independent_backtesting.py [hunk 3]
```diff
@@ -41,7 +41,8 @@ def __init__(self, config,
                  tentacles_setup_config,
                  backtesting_files,
                  data_file_path=backtesting_constants.BACKTESTING_FILE_PATH,
-                 run_on_common_part_only=True):
+                 run_on_common_part_only=True,
+                 disable_join_timeout=False):
```
Comment (GuillaumeDSM) [inline octobot/backtesting/independent_backtesting.py:45]: I think it would be clearer to name it "enable_timeout"

File: octobot/cli.py (hunk 2 above)
Comment (Herklos) [reply octobot/cli.py:239]: :+1:

File: octobot/backtesting/independent_backtesting.py (hunk 3 above)
Comment (Herklos) [reply octobot/backtesting/independent_backtesting.py:45]: Yes

File: octobot/backtesting/independent_backtesting.py [hunk 4]
```diff
@@ -56,6 +57,8 @@ def __init__(self, config,
         self._init_default_config_values()
         self.stopped = False
         self.post_backtesting_task = None
+        self.join_backtesting_timeout = backtesting_constants.BACKTESTING_DEFAULT_JOIN_TIMEOUT \
+            if enable_join_timeout else None
```
Comment (GuillaumeDSM) [inline octobot/backtesting/independent_backtesting.py:61]: 👍

File: octobot/cli.py [hunk 5]
```diff
@@ -232,6 +234,10 @@ def octobot_parser(parser):
                         help='On multiple files backtesting: run on the whole available data instead of the '
                              'common part only (default behavior).',
                         action='store_true')
+    parser.add_argument('-ebt', '--enable-backtesting-timeout',
+                        help='To disable when running long backtesting : disable backtesting watcher timeout '
+                             'that prevent backtesting from being interrupted during execution.',
```
Comment (GuillaumeDSM) [inline octobot/cli.py:239]: Limit the backtesting run to backtesting_constants.BACKTESTING_DEFAULT_JOIN_TIMEOUT /60 minutes.

File: octobot/cli.py [hunk 6]
```diff
@@ -232,6 +234,11 @@ def octobot_parser(parser):
                         help='On multiple files backtesting: run on the whole available data instead of the '
                              'common part only (default behavior).',
                         action='store_true')
+    parser.add_argument('-ebt', '--enable-backtesting-timeout',
+                        help='To disable when running long backtesting : disable backtesting watcher timeout '
```
Comment (GuillaumeDSM) [inline octobot/cli.py:238]: We should remove this first part of the help, there is nothing to disable as it's not enabled by default.

**Review by GuillaumeDSM — APPROVED** [review]
👍

**Pull request merged**